### ISSUE TITLE
Feature: detect return values other than 0-225 (like strings) in functions when used via variables

### ISSUE THREAD

**Issue by mika** [koalaman/shellcheck#1060]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
foo() {
  local file="/usr/bin/foo"
  return "$file"
}
foo
```

#### Here's what shellcheck currently says:

`No issues detected!`

#### Here's what I wanted or expected to see:

`SC2152: Can only return 0-255. Other data should be written to stdout.`

JFTR, shellcheck detects it when used without a variable but a plain string already:

```sh
#!/bin/bash
foo() {
  return bar                                                                                                                                                                                                                                 
}
foo
```

This is reported as `SC2152: Can only return 0-255. Other data should be written to stdout.`.

Thanks for shellcheck!
